src/algorithms/peps_opt.jl

```diff
@@ -45,12 +45,24 @@ function fixedpoint(
     ψ₀::InfinitePEPS{T}, H, alg::PEPSOptimize, env₀::CTMRGEnv=CTMRGEnv(ψ₀; Venv=field(T)^20)
 ) where {T}
     (peps, env), E, ∂E, info = optimize(
-        x -> ctmrg_gradient(x, H, alg),
-        (ψ₀, env₀),
-        alg.optimizer;
-        retract=my_retract,
-        inner=my_inner,
-    )
+        (ψ₀, env₀), alg.optimizer; retract=my_retract, inner=my_inner
+    ) do (peps, envs)
+        E, g = withgradient(peps) do ψ
+            envs´ = hook_pullback(
+                leading_boundary,
+                ψ,
+                alg.boundary_alg,
+                envs;
+                alg_rrule=alg.gradient_alg,
+            )
+            ignore_derivatives() do
+                alg.reuse_env && (envs.edges .= envs´.edges; envs.corners .= envs´.corners)
```

Comment:
```suggestion
                alg.reuse_env && update!(envs, envs´)
```
@leburgel , can you refactor this into a separate function?